bftengine/src/communication/PlainUDPCommunication.cpp

```diff
@@ -340,24 +336,10 @@ class PlainUDPCommunication::PlainUdpImpl {
   void
   stopRecvThread() {
     //LOG_DEBUG("Stopping the receiving thread..");
-    threadJoin(recvThreadRef);
+    recvThreadRef->join();
```

Comment:
recvThreadRef can be nullptr or the wrapped thread can be non joinable, if the Start() method was not called. I suggest to check for the pointer validity and thread being joinable.